v-next/hardhat/src/internal/cli/init/init.ts

```diff
@@ -529,3 +528,34 @@ function showStarOnGitHubMessage() {
   console.log();
   console.log(chalk.cyan("     https://github.com/NomicFoundation/hardhat"));
 }
+
+// NOTE: This function is exported for testing purposes only.
```

Comment:
Nit: I don't think you need the comment. I think our internal/external folder structure covers us.
We could look at adding better eslint rules for unused exports, but I think that is a general thing.